### PR TITLE
Release candidate for v1.0.0

### PR DESCRIPTION
**Pull request by beevital**
OAS based connector resource

PR size note:
Don't review open-api-spec.json - it contains most part of changes.

### CODE REVIEW

File: fivetran/config-changes.txt [hunk 1]
```diff
@@ -0,0 +1,158 @@
+- Added field `fivetran_connector.config.account_name` for services: `talkdesk`.
```
Comment (Fivetran-RomanKolesnikov) [inline fivetran/config-changes.txt:1]: Is it temporary file with generated changelog? should it be committed?

File: fivetran/helpers.go [hunk 2]
```diff
@@ -123,13 +169,13 @@ func intPointerToStr(i *int) string {
...
 
 // xStrXInterface receives a []string and returns a []interface{}
-func xStrXInterface(xs []string) []interface{} {
-	xi := make([]interface{}, len(xs))
-	for i, v := range xs {
-		xi[i] = v
-	}
-	return xi
-}
+// func xStrXInterface(xs []string) []interface{} {
```
Comment (Fivetran-RomanKolesnikov) [inline fivetran/helpers.go:172]: Is it commented code needed?

File: fivetran/tests/mock/resource_connector_conflicting_fields_mapping_test.go [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+		func(req *http.Request) (*http.Response, error) {
+			return fivetranSuccessResponse(t, req, http.StatusOK, "Success", connectorConflictingMappingMockData), nil
+		},
+	)
+
+	connectorConflictingMockPostHandler = mockClient.When(http.MethodPost, "/v1/connectors").ThenCall(
+		func(req *http.Request) (*http.Response, error) {
+			//body := requestBodyToJson(t, req)
+
+			//assertKeyExists(t, body, "config")
```
Comment (Fivetran-RomanKolesnikov) [inline fivetran/tests/mock/resource_connector_conflicting_fields_mapping_test.go:103]: Do we need commented code in this file?

**Review by Fivetran-RomanKolesnikov — APPROVED** [review]
Looks good!

File: fivetran/connector_common.go [hunk 4]
```diff
@@ -51,12 +125,12 @@ func getConnectorSchema(readonly bool, version int) map[string]*schema.Schema {
...
 			ForceNew:    !readonly,
 			Computed:    readonly,
-			Description: "The connector type name within the Fivetran system",
+			Description: "The connector type name within the Fivetran system.",
 		},
 		"destination_schema": getConnectorDestinationSchema(readonly),
 
 		// Config
-		"config": getConnectorSchemaConfig(readonly, version),
+		"config": connectorSchemaConfig(readonly),
```
Comment (LukaDevic) [inline fivetran/connector_common.go:133]: If we have "get" in the name that clearly indicates we are calling a function, without it it looks like we are initialising an object, which isn't the case.
My suggestion is to keep "getConnectorSchemaConfig" name.

File: fivetran/connector_common.go [hunk 5]
```diff
@@ -274,35 +339,58 @@ func getConnectorRead(currentConfig *[]interface{}, resp fivetran.ConnectorCusto
 		mapAddStr(msi, "paused", boolPointerToStr(resp.Data.Paused))
 		mapAddStr(msi, "pause_after_trial", boolPointerToStr(resp.Data.PauseAfterTrial))
 
-		mapAddXInterface(msi, "status", getConnectorReadStatus(&resp))
+		mapAddXInterface(msi, "status", connectorReadStatus(&resp))
```
Comment (LukaDevic) [inline fivetran/connector_common.go:342]: Same here as mine previous comment.

File: fivetran/connector_common.go [hunk 6]
```diff
@@ -274,35 +339,58 @@ func getConnectorRead(currentConfig *[]interface{}, resp fivetran.ConnectorCusto
...
 	}
-	upstreamConfig := getConnectorReadCustomConfig(&resp, currentConfig)
+	upstreamConfig := connectorReadCustomConfig(&resp, currentConfig, resp.Data.Service)
```
Comment (LukaDevic) [inline fivetran/connector_common.go:344]: Same as my other comments

File: fivetran/connector_common.go [hunk 7]
```diff
@@ -274,35 +339,58 @@ func getConnectorRead(currentConfig *[]interface{}, resp fivetran.ConnectorCusto
...
+	}
+
+	destination_schema[0] = ds
+	return destination_schema, nil
 }
 
 // resourceConnectorReadStatus receives a *fivetran.ConnectorDetailsResponse and returns a []interface{}
 // containing the data type accepted by the "status" list.
-func getConnectorReadStatus(resp *fivetran.ConnectorCustomDetailsResponse) []interface{} {
+func connectorReadStatus(resp *fivetran.ConnectorCustomDetailsResponse) []interface{} {
```
Comment (LukaDevic) [inline fivetran/connector_common.go:378]: Same as other comments

File: fivetran/connector_common.go [hunk 8]
```diff
@@ -274,35 +339,58 @@ func getConnectorRead(currentConfig *[]interface{}, resp fivetran.ConnectorCusto
...
 	status := make([]interface{}, 1)
 
 	s := make(map[string]interface{})
 	mapAddStr(s, "setup_state", resp.Data.Status.SetupState)
 	mapAddStr(s, "sync_state", resp.Data.Status.SyncState)
 	mapAddStr(s, "update_state", resp.Data.Status.UpdateState)
 	mapAddStr(s, "is_historical_sync", boolPointerToStr(resp.Data.Status.IsHistoricalSync))
-	mapAddXInterface(s, "tasks", getConnectorReadStatusFlattenTasks(resp))
-	mapAddXInterface(s, "warnings", getConnectorReadStatusFlattenWarnings(resp))
+	mapAddXInterface(s, "tasks", connectorReadStatusFlattenTasks(resp))
```
Comment (LukaDevic) [inline fivetran/connector_common.go:386]: Same as other comments

File: fivetran/connector_common.go [hunk 9]
```diff
@@ -274,35 +339,58 @@ func getConnectorRead(currentConfig *[]interface{}, resp fivetran.ConnectorCusto
...
+	mapAddXInterface(s, "warnings", connectorReadStatusFlattenWarnings(resp))
```
Comment (LukaDevic) [inline fivetran/connector_common.go:387]: Same as other comments

File: fivetran/connector_common.go [hunk 10]
```diff
@@ -274,35 +339,58 @@ func getConnectorRead(currentConfig *[]interface{}, resp fivetran.ConnectorCusto
...
 	status[0] = s
 
 	return status
 }
 
-func getConnectorReadStatusFlattenTasks(resp *fivetran.ConnectorCustomDetailsResponse) []interface{} {
+func connectorReadStatusFlattenTasks(resp *fivetran.ConnectorCustomDetailsResponse) []interface{} {
```
Comment (LukaDevic) [inline fivetran/connector_common.go:393]: Same as other comments

File: fivetran/connector_common.go [hunk 11]
```diff
@@ -319,7 +407,7 @@ func getConnectorReadStatusFlattenTasks(resp *fivetran.ConnectorCustomDetailsRes
 	return tasks
 }
 
-func getConnectorReadStatusFlattenWarnings(resp *fivetran.ConnectorCustomDetailsResponse) []interface{} {
+func connectorReadStatusFlattenWarnings(resp *fivetran.ConnectorCustomDetailsResponse) []interface{} {
```
Comment (LukaDevic) [inline fivetran/connector_common.go:410]: Same as other comments

File: fivetran/data_source_connector.go [hunk 12]
```diff
@@ -12,28 +12,33 @@ import (
 func dataSourceConnector() *schema.Resource {
 	return &schema.Resource{
 		ReadContext: dataSourceConnectorRead,
-		Schema:      getConnectorSchema(true, 0),
+		Schema:      connectorSchema(true, 0),
```
Comment (LukaDevic) [inline fivetran/data_source_connector.go:15]: Same as other comments

File: fivetran/resource_connector.go [hunk 13]
```diff
@@ -17,22 +17,45 @@ func resourceConnector() *schema.Resource {
 		UpdateContext: resourceConnectorUpdate,
 		DeleteContext: resourceConnectorDelete,
 		Importer:      &schema.ResourceImporter{StateContext: schema.ImportStatePassthroughContext},
-		Schema:        getConnectorSchema(false, 1),
-		SchemaVersion: 1,
+		Schema:        connectorSchema(false, 2),
```
Comment (LukaDevic) [inline fivetran/resource_connector.go:20]: Same as other comments

File: fivetran/tests/mock/resource_connector_conflicting_fields_mapping_test.go [hunk 14]
```diff
@@ -0,0 +1,275 @@
...
+
+	connectorConflictingMockDelete = mockClient.When(http.MethodDelete, "/v1/connectors/connector_id").ThenCall(
+		func(req *http.Request) (*http.Response, error) {
+			connectorConflictingMappingMockData = nil
+			return fivetranSuccessResponse(t, req, http.StatusOK, "Success", connectorConflictingMappingMockData), nil
+		},
+	)
+}
+
+//func testConflictingField(tfConfig, jsonResponse string, checkFunc resource.TestCheckFunc, )
```
Comment (LukaDevic) [inline fivetran/tests/mock/resource_connector_conflicting_fields_mapping_test.go:120]: Commented code

File: utils/generate_connector_config.go [hunk 15]
```diff
@@ -0,0 +1,394 @@
+package main
+
```
Comment (LukaDevic) [inline utils/generate_connector_config.go:2]: File name is in the format of functions names, maybe name should be something like "connector_config_service"

File: utils/generate_connector_config.go [hunk 16]
```diff
@@ -0,0 +1,394 @@
...
+					fmt.Println("INFO: field " + name + " ignored as inconsistent. Service: " + service + ".")
+				}
+			}
+		}
+	}
+	return updated, changeLog
+}
+
+func fieldCouldBeIncluded(field fivetran.ConfigField) bool {
+
```
Comment (LukaDevic) [inline utils/generate_connector_config.go:153]: Maybe this empty row could be deleted

**Review by LukaDevic — APPROVED** [review]
Looks good with comments.

File: utils/generate_connector_config.go (hunk 15 above)
Comment (beevital) [reply utils/generate_connector_config.go:2]: we can think about this file as a function used by go:generate utility =)
But we can use any name for it. If you suggest connector_config_service - let it be

File: fivetran/connector_common.go (hunk 6 above)
Comment (beevital) [reply fivetran/connector_common.go:344]: this name means that this function is related to connector resource/datasource and it Reads custom config response - there's no need to add get

File: fivetran/connector_common.go (hunk 7 above)
Comment (beevital) [reply fivetran/connector_common.go:378]: same here - we have Read verb in name

File: fivetran/connector_common.go (hunk 5 above)
Comment (beevital) [reply fivetran/connector_common.go:342]: connector**Read**Status mean action already

File: fivetran/data_source_connector.go (hunk 12 above)
Comment (beevital) [reply fivetran/data_source_connector.go]: updated

**Pull request merged**